src/openai.ts

```diff
@@ -29,15 +29,17 @@ import type {
   Translation,
   TranslationOptions,
 } from "./types.ts";
-import { basename } from "https://deno.land/std@0.185.0/path/mod.ts";
+import { basename, join } from "https://deno.land/std@0.185.0/path/mod.ts";
 
-const baseUrl = "https://api.openai.com/v1";
+const defaultBaseUrl = "https://api.openai.com/v1";
 
 export class OpenAI {
   #privateKey: string;
+  #baseUrl: URL;
 
-  constructor(privateKey: string) {
+  constructor(privateKey: string, baseUrl?: string) {
```

Comment:
Could we change this to a "bag of options". I find it likely that we may want to add more options in the future and would hate to make a breaking change to the API.
```suggestion
  constructor(privateKey: string, options?: { baseUrl?: string }) {
```